lib/saluki-components/src/transforms/dogstatsd_prefix_filter/mod.rs

```diff
@@ -0,0 +1,349 @@
+        data.sort();
+
+        // Only keep values with unique prefixes.
+        if match_prefix && !data.is_empty() {
+            let mut i = 0;
+            for j in 1..data.len() {
+                if !data[j].starts_with(&data[i]) {
+                    i += 1;
+                    data[i] = data[j].clone(); // Move item to next position
+                }
+            }
+            data.truncate(i + 1);
+        }
+
+        Blocklist { data, match_prefix }
+    }
+
+    fn contains(&self, name: &str) -> bool {
+        if self.data.is_empty() {
+            return false;
+        }
+
+        let i = self.data.binary_search(&name.to_string());
+
+        // Check for prefix match when match_prefix is true
+        if self.match_prefix {
+            let index = i.unwrap_or_else(|idx| idx);
+            if index > 0 && name.starts_with(&self.data[index - 1]) {
+                return true;
+            }
+        }
+
+        if let Ok(index) = i {
+            return name == self.data[index];
+        }
+
+        false
+    }
+}
+
+#[derive(Debug)]
+pub struct DogstatsDPrefixFilter {
+    metric_prefix: String,
+
+    metric_prefix_blacklist: Vec<String>,
+
+    blocklist: Blocklist,
+}
+
+impl DogstatsDPrefixFilter {
+    fn enrich_metric(&self, mut metric: Metric) -> Option<Metric> {
+        let metric_name = metric.context().name().deref();
+
+        if self.metric_prefix.is_empty() {
+            for s in &self.blocklist.data {
+                if s == metric_name || self.blocklist.match_prefix && metric_name.starts_with(s) {
+                    return None;
+                }
+            }
+        } else {
+            // Enrich metric with prefix if prefix is allowed.
+            let new_metric_name = if self.is_excluded(metric_name) {
+                metric.context().name().clone()
+            } else {
+                MetaString::from(format!("{}{}", self.metric_prefix, metric_name))
+            };
+
+            if self.blocklist.contains(&new_metric_name) {
+                debug!("Metric {} excluded due to blocklist.", new_metric_name);
+                return None;
+            }
+
+            // Update metric with new name.
+            let new_context = metric.context().with_name(new_metric_name);
+            let existing_context = metric.context_mut();
+            *existing_context = new_context;
+        }
+        Some(metric)
+    }
+
+    fn is_excluded(&self, metric_name: &str) -> bool {
+        !self.metric_prefix.is_empty()
+            && self
+                .metric_prefix_blacklist
+                .iter()
+                .any(|prefix| metric_name.starts_with(prefix))
+    }
+}
+
+#[async_trait]
+impl Transform for DogstatsDPrefixFilter {
+    async fn run(mut self: Box<Self>, mut context: TransformContext) -> Result<(), GenericError> {
+        let mut health = context.take_health_handle();
+        health.mark_ready();
+
+        debug!("Agent metric name filter transform started.");
```

Comment:
I would change this from `Agent metric name filter` to `DogStatsD Prefix Filter`, just to make sure we're consistent with how other components log their human-friendly name.

Reply:
(And then also update down below as well.)